src/filecache.c

```diff
@@ -768,6 +768,12 @@ static void get_fresh_fd(filecache_t *cache,
                 funcname, path, lg, sz, atime, lsu);
 
             stat_cache_delete(cache, path, NULL);
+            // If we delete this file locally but it really exists, then a subsequent progressive
+            // propfind will not indicate that it exists, and this binding will continue
+            // to think it doesn't exist. Force a complete propfind next time on the
```

Comment:
Slightly concerned about how many additional complete propfinds we might be generating with this.  Have we measured this yet?

Reply:
Point taken. Since it only occurs when we respond to a 404/410/PFT, it should be rare enough. I'll gather some data.

Reply:
I share the concerns of @kf6nux. Can we discuss this when I'm back in the office next week, or is this urgent?

Reply:
Suppose at attempt is made for file /a/b/c/d.txt.
If the Propfind finds the file locally, but on the GET, the fileserver indicates the file is missing, we get an "unexpected 404"
If on the GET the fileserver returns 410 (likely means file is not in permanent storage), we get a 410.
If the error message is "partial file transfer", we detect that here.
In these cases we will set the last updated value on the parent directory c (parent of d.txt) to 0, and the next time this binding accesses that directory, it will be a complete propfind.
This is at least the intent. Is this not what you think the implementation will do?
I assume that this is rare enough overall that it will not significantly impact the system. I am collecting stats now.
We could assume that a 404 or a 410 from the fileserver really means that the file doesn't exist there, so a complete propfind to ascertain that is unnecessary. Then we would only reset stat_cache_updated_children on "partial file transfer" where it is most necessary.